RIOT-OS/RIOT
# tests: introduce dedicated `gnrc_rpl_srh` test

### Contribution description
This introduces a dedicated test application for GNRC's RPL source router header implementation. It utilizes `scapy` for the generation of the routing headers. It also moves the unittests from the unittests application into this application.

### Testing procedure
Compile `tests/gnrc_rpl_srh` and flash it to a board of your choice. Use `sudo make test` to run (due to the required Ethernet frame construction with the tap interface you need root permissions). The tests might hang and fail due to `scapy`'s sniffer sometimes not picking up the expected packet. Maybe someone can help me with syncing the main thread and the sniffer thread of the python script here?

### Issues/PRs references
Depends on #10242, #10244, and ~~#10300~~(merged) and their dependencies:

![PR dependencies](http://page.mi.fu-berlin.de/mlenders/ipv6_rework.svg)

Addresses #8941 partly.
